File: backend/src/main/java/com/mindflow/service/NoteService.java
package com.mindflow.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.mindflow.entity.Note;
import com.mindflow.vo.InterviewQuestionVO;

import java.util.List;
import java.util.Map;

/**
 * 笔记服务接口
 */
public interface NoteService extends IService<Note> {

    /** 创建笔记（含 AI 自动总结） */
    Map<String, Object> createNote(String content, String contentType);

    /** 分页查询笔记列表 */
    Map<String, Object> getNoteList(int pageNum, int pageSize, String category, String keyword);

    /** 笔记详情 */
    Map<String, Object> getNoteDetail(Long noteId);

    /** 删除笔记 */
    void deleteNote(Long noteId);

    /** 生成面试题 */
    List<InterviewQuestionVO> generateInterviewQuestions(Long noteId);

    /** 获取用户所有分类 */
    List<String> getCategories();

    /** 获取用户所有标签 */
    List<String> getTags();
}

File: frontend/lib/services/note_service.dart
import 'api_service.dart';
import '../models/note.dart';

class NoteService {
  final ApiService _api = ApiService();

  Future<Map<String, dynamic>> createNote(String content) async {
    final resp = await _api.dio.post('/api/note/create', data: {
      'content': content,
      'contentType': 'text',
    });
    return resp.data['data'];
  }

  Future<Map<String, dynamic>> getNoteList({
    int pageNum = 1,
    int pageSize = 10,
    String? category,
    String? keyword,
  }) async {
    final params = <String, dynamic>{
      'pageNum': pageNum,
      'pageSize': pageSize,
    };
    if (category != null && category.isNotEmpty) params['category'] = category;
    if (keyword != null && keyword.isNotEmpty) params['keyword'] = keyword;

    final resp = await _api.dio.get('/api/note/list', queryParameters: params);
    final data = resp.data['data'];
    return {
      'list': (data['list'] as List).map((e) => Note.fromJson(e)).toList(),
      'total': data['total'] ?? 0,
      'pageNum': data['pageNum'] ?? 1,
      'pageSize': data['pageSize'] ?? 10,
    };
  }

  Future<NoteDetail> getNoteDetail(int id) async {
    final resp = await _api.dio.get('/api/note/detail/$id');
    return NoteDetail.fromJson(resp.data['data']['noteInfo']);
  }

  Future<void> deleteNote(int id) async {
    await _api.dio.delete('/api/note/delete/$id');
  }

  Future<List<String>> getCategories() async {
    final resp = await _api.dio.get('/api/note/categories');
    return List<String>.from(resp.data['data'] ?? []);
  }

  Future<List<String>> getTags() async {
    final resp = await _api.dio.get('/api/note/tags');
    return List<String>.from(resp.data['data'] ?? []);
  }
}